src/charm.py

```diff
@@ -38,21 +33,26 @@ def __init__(self, *args: typing.Any) -> None:
             args: passthrough to CharmBase.
         """
         super().__init__(*args)
-        self._databases = Databases(charm=self)
         self._secret_storage = SecretStorage(charm=self)
-
+        self._database_requirers = Databases.make_database_requirers(self)
```

Comment:
I'm guessing this is a static method, should this just be top level function in the module?

Reply:
Updated, thanks